### PR TITLE
feat(SuccessFeedback): new component

### PR DESCRIPTION
**Pull request by anabelengp**
Depends on #343 

SuccessFeedback is a component that can be rendered along with other components. It doesn't include any margin with upper or lower components.

I just refactored some code in `feedback-screen.tsx -> feedback.tsx` to share a couple of methods to render the code that is common in feedback-screen and success-feedback. They basically differ in that the `screen` one uses a layout to compose the entire page for both mobile and desktop. 

This PR also manages the problem of disabling animations in CSS related to https://jira.tid.es/browse/CHECKOUT-2512

### CODE REVIEW

File: src/__stories__/success-feedback-story.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    return (
+        <Stack space={isTabletOrSmaller ? 24 : 40}>
+            {isTabletOrSmaller ? successFeedback : <ResponsiveLayout>{successFeedback}</ResponsiveLayout>}
+            <ResponsiveLayout>
+                <Stack space={16}>
+                    <Placeholder />
+                    <Placeholder />
+                </Stack>
+            </ResponsiveLayout>
+        </Stack>
```
Comment (anabelengp) [inline src/__stories__/success-feedback-story.tsx:89]: Is this the expected usage in [here](https://app.zeplin.io/project/5d653c69f828bf7299c551c1/screen/613a10018b0235b8a976eb14)
Section: Header Success Feedback

Comment (anabelengp) [reply src/__stories__/success-feedback-story.tsx:89]: (see the uploaded screenshots to compare)

File: src/feedback.tsx [hunk 2]
```diff
@@ -324,3 +379,62 @@ export const InfoFeedbackScreen: React.FC<FeedbackProps> = (props) => {
...
+            secondaryButton,
+            link,
+        },
+        classes
+    );
+
+    return (
+        <ThemeVariant isInverse>
+            {isTabletOrSmaller ? (
+                <ResponsiveLayout className={classes.background}>
```
Comment (anabelengp) [inline src/feedback.tsx:423]: ResponsiveLayout included here to implement this:
image

**Review by pladaria — APPROVED** [review]

File: src/__stories__/success-feedback-story.tsx (hunk 1 above)
Comment (pladaria) [reply src/__stories__/success-feedback-story.tsx:89]: lgtm

**Review by HectorSancho — APPROVED** [review]

**Review by atabel — APPROVED** [review]

File: src/feedback.tsx [hunk 3]
```diff
@@ -90,31 +79,28 @@ const useStyles = createUseStyles((theme) => ({
...
-    },
-
     innerContainer: {
         textAlign: 'left',
         padding: '64px 8px 0',
     },
 
     feedbackData: {
-        animation: animateText(theme.platformOverrides),
-        opacity: initialTextOpacity(theme.platformOverrides),
```
Comment (anabelengp) [inline src/feedback.tsx:104]: This change is related to https://github.com/Telefonica/webapp/pull/907. When animations/transitions are disabled by CSS, the feedback text keeps `opacity: 0, translate(0,20px)`. We now manage the transition from React in order to be sure that the final state will always be reached.

File: src/feedback.tsx [hunk 4]
```diff
@@ -154,13 +140,34 @@ const useHapticFeedback = (type?: HapticFeedback) => {
...
+ * even if animations/transitions are disabled by css
+ * (CSSTransition does not work here when running in Storybook)
+ * @returns boolean whether the transition should start
+ */
+const useAppearStatus = (): boolean => {
+    // A state to manage the text animation with transitions instead of keyframes
+    const [appear, setAppear] = React.useState(false);
+
+    React.useEffect(() => {
+        setTimeout(() => {
```
Comment (atabel) [inline src/feedback.tsx:154]: maybe a React.useLayoutEffect + requestAnimationFrame is better here

**Review by yceballost — APPROVED** [review]

File: src/__stories__/success-feedback-story.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            }
+            primaryButton={withButtons ? primaryButton : undefined}
+            secondaryButton={withButtons ? secondaryButton : undefined}
+            imageUrl={imageUrl ?? undefined}
+            imageFit={imageFit}
+        />
+    );
+    return (
+        <Stack space={isTabletOrSmaller ? 24 : 40}>
+            {isTabletOrSmaller ? successFeedback : <ResponsiveLayout>{successFeedback}</ResponsiveLayout>}
```
Comment (pladaria) [inline src/__stories__/success-feedback-story.tsx:82]: I find this ugly but I guess that's the best we can get :/

**Review by pladaria — APPROVED** [review]
:+1:

**Review by atabel — APPROVED** [review]

**Pull request merged**